src/librustdoc/html/static/js/search.js

```diff
@@ -2853,6 +2831,108 @@ ${item.displayPath}<span class="${type}">${name}</span>\
+     */
+    function buildFunctionTypeFingerprint(type, output, fps) {
+
+        let input = type.id;
+        // All forms of `[]` get collapsed down to one thing in the bloom filter.
+        // Differentiating between arrays and slices, if the user asks for it, is
+        // still done in the matching algorithm.
+        if (input === typeNameIdOfArray || input === typeNameIdOfSlice) {
+            input = typeNameIdOfArrayOrSlice;
+        }
+        const fxScram = k => {
+            // https://docs.rs/rustc-hash/1.1.0/src/rustc_hash/lib.rs.html#60
+            return Math.imul((k << 5) | (k >> 27), 0x9e3779b9);
+        };
+        const murmurScram = k => {
+            // https://en.wikipedia.org/wiki/MurmurHash
+            k = Math.imul(k, 0xcc9e2d51);
+            return Math.imul((k << 15) | (k >> 17), 0x1b873593);
+        };
+        if (input !== null) {
+            const h0a = fxScram(input);
+            const h0b = murmurScram(input);
+            const h1a = fxScram(479001599 ^ input);
+            const h1b = murmurScram(479001599 ^ input);
+            const h2a = fxScram(433494437 ^ input);
+            const h2b = murmurScram(433494437 ^ input);
+            output[0] |= (1 << (h0a % 32)) | (1 << (h0b % 32));
+            output[1] |= (1 << (h1a % 32)) | (1 << (h1b % 32));
+            output[2] |= (1 << (h2a % 32)) | (1 << (h2b % 32));
+            fps.add(input);
+        }
+        for (const g of type.generics) {
+            buildFunctionTypeFingerprint(g, output, fps);
+        }
+        const fb = {
+            id: null,
+            ty: 0,
+            generics: [],
+            bindings: new Map(),
+        };
+        for (const [k, v] of type.bindings.entries()) {
+            fb.id = k;
+            fb.generics = v;
+            buildFunctionTypeFingerprint(fb, output, fps);
+        }
+        output[3] = fps.size;
+    }
+
+    /**
+     * Compare the query fingerprint with the function fingerprint.
+     *
+     * @param {{number}} fullId - The function
+     * @param {{Uint32Array}} queryFingerprint - The query
+     * @returns {number|false} - False if non-match, number if distance
+     *                           This function might return 0!
+     */
+    function compareTypeFingerprints(fullId, queryFingerprint) {
+        const fh0 = functionTypeFingerprint[fullId * 4];
+        const fh1 = functionTypeFingerprint[(fullId * 4) + 1];
+        const fh2 = functionTypeFingerprint[(fullId * 4) + 2];
+        const qh0 = queryFingerprint[0];
+        const qh1 = queryFingerprint[1];
+        const qh2 = queryFingerprint[2];
+        // Approximate set intersection with bloom filters.
+        // This can be larger than reality, not smaller, because hashes have
+        // the property that if they've got the same value, they hash to the
+        // same thing. False positives exist, but not false negatives.
+        const [in0, in1, in2] = [fh0 & qh0, fh1 & qh1, fh2 & qh2];
```

Comment:
Is there any reason to write it like this? You created each variable separately above after all.